File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/presentation/adapter/UserAdapter.java
package com.kalashnyk.denys.samplemvvmapp.presentation.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.kalashnyk.denys.samplemvvmapp.R;
import com.kalashnyk.denys.samplemvvmapp.presentation.base.BaseAdapter;
import com.kalashnyk.denys.samplemvvmapp.presentation.item.IUserItemClickListener;
import com.kalashnyk.denys.samplemvvmapp.presentation.item.UserViewHolder;
import com.kalashnyk.denys.samplemvvmapp.repository.database.entity.UserEntity;

import java.util.List;

public class UserAdapter extends BaseAdapter<UserViewHolder, UserEntity, IUserItemClickListener> {

    private List<UserEntity> mUsers;
    private Context mContext;

    public UserAdapter(Context context, List<UserEntity> list) {
        super(context, list);
        mUsers = list;
        mContext = context;
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new UserViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_user, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder userViewHolder, int i) {
        super.onBindViewHolder(userViewHolder, i);
        userViewHolder.bind(mUsers.get(i), mItemClickListener);
    }

    @Override
    public int getItemCount() {
        return mUsers.size();
    }
}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/presentation/adapter/UserAdapter.kt
package com.kalashnyk.denys.kotlinsample.presentation.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import com.kalashnyk.denys.kotlinsample.R
import com.kalashnyk.denys.kotlinsample.presentation.base.BaseAdapter
import com.kalashnyk.denys.kotlinsample.presentation.item.IUserItemClickListener
import com.kalashnyk.denys.kotlinsample.presentation.item.UserViewHolder
import com.kalashnyk.denys.kotlinsample.repository.database.entity.UserEntity

class UserAdapter(private val context: Context, private val users: List<UserEntity>, private val listener: IUserItemClickListener<UserEntity>) :
    BaseAdapter<UserViewHolder, UserEntity, IUserItemClickListener<UserEntity>>(users as MutableList<UserEntity>, listener) {

    override fun getItemCount(): Int {
        return users.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        return UserViewHolder(LayoutInflater.from(context).inflate(R.layout.item_user, parent, false))
    }

    override fun onBindViewHolder(userViewHolder: UserViewHolder, i: Int) {
        super.onBindViewHolder(userViewHolder, i)
        userViewHolder.bind(users[i], listener)
    }
}
